programs/fileio.c

```diff
@@ -862,6 +862,193 @@ int FIO_compressFilename(const char* dstFileName, const char* srcFileName,
+                    info->canComputeDecompSize = 0;
+                }
+            }
+
+            /* skip the rest of the blocks in the frame */
+            do{
+                BYTE blockHeaderBuffer[3];
+                U32 blockHeader;
+                int blockSize;
+                readBytes = fread(blockHeaderBuffer, 1, 3, srcFile);
+                if(readBytes != 3){
+                    DISPLAY("There was a problem reading the block header\n");
+                    exit(1);
+                }
+                blockHeader = MEM_readLE24(blockHeaderBuffer);
+                lastBlock = blockHeader & 1;
+                blockSize = (blockHeader - (blockHeader & 7)) >> 3;
+                fseek(srcFile, blockSize, SEEK_CUR);
+            }while(lastBlock != 1);
+            {
+                /* check if checksum is used */
+                int contentChecksumFlag = (frameHeaderDescriptor & (1 << 2)) >> 2;
+                if(contentChecksumFlag){
+                    info->usesCheck = 1;
+                }
+                if(contentChecksumFlag){
+                    fseek(srcFile, 4, SEEK_CUR);
+                }
+            }
+        }
+        else if(magicNumber==ZSTD_MAGIC_SKIPPABLE_START){
+            BYTE frameSizeBuffer[4];
+            long frameSize;
+            size_t readBytes = fread(frameSizeBuffer, 1, 4, srcFile);
+            info->numSkippableFrames++;
+            if(readBytes != 4){
+                DISPLAY("There was an error reading skippable frame size");
+                exit(1);
+            }
+            frameSize = MEM_readLE32(frameSizeBuffer);
+            fseek(srcFile, frameSize, SEEK_CUR);
+        }
+
+    }
+    return 0;
+}
+void displayInfo(const char* inFileName, fileInfo_t* info, int displayLevel){
+    double compressedSizeMB = (double)info->compressedSize/(1 MB);
+    double decompressedSizeMB = (double)info->decompressedSize/(1 MB);
+
+    if(displayLevel<=2){
+        if(info->usesCheck && info->canComputeDecompSize){
+            DISPLAY("Skippable  Non-Skippable  Compressed  Uncompressed  Ratio  Check  Filename\n");
+            DISPLAY("%9d  %13d  %7.2f MB    %7.2f MB  %5.3f  XXH64  %s\n",
+                    info->numSkippableFrames, info->numActualFrames, compressedSizeMB, decompressedSizeMB,
+                    compressedSizeMB/decompressedSizeMB, inFileName);
+        }
+        else if(!info->usesCheck){
+            DISPLAY("Skippable  Non-Skippable  Compressed  Uncompressed  Ratio  Check  Filename\n");
+            DISPLAY("%9d  %13d  %7.2f MB    %7.2f MB  %5.3f  %s\n",
+                    info->numSkippableFrames, info->numActualFrames, compressedSizeMB, decompressedSizeMB,
+                    compressedSizeMB/decompressedSizeMB, inFileName);
+        }
+        else if(!info->canComputeDecompSize){
+            DISPLAY("Skippable  Non-Skippable  Compressed  Uncompressed  Ratio  Check  Filename\n");
+            DISPLAY("%9d  %13d  %7.2f MB      XXH64  %s\n",
```

Comment:
It will probably simplify your branching if you make `XXH64` a `%s` argument.
Something like : 
`const char checkString[] = info->usesCheck ? " XXH64 " : "  None ";`